setup.py

```diff
@@ -298,9 +323,17 @@ def cmake(build_dir: str = "", install_prefix: str = "./nvfuser"):
     print(f"Configuring CMake with {' '.join(cmd_str)}")
     subprocess.check_call(cmd_str)
 
+    max_jobs = multiprocessing.cpu_count()
+    mem_gb_per_task = 3  # Currently compilation of nvFuser souce code takes ~3GB of memory per task, we should adjust this value if it changes in the future.
+    available_mem = get_available_memory_gb()
+    if available_mem > 0:
+        max_jobs_mem = int(available_mem / mem_gb_per_task)
+        max_jobs = min(max_jobs, max_jobs_mem)
+
     if not CMAKE_ONLY:
         # build binary
-        max_jobs = os.getenv("MAX_JOBS", str(multiprocessing.cpu_count()))
+        max_jobs = os.getenv("MAX_JOBS", str(max_jobs))
+        print(f"Using {max_jobs} jobs for compilation")
```

Comment:
I thought we have a logging system. Looks like I was just hallucinating.

Reply:
We do, except that we defined a `logger` and just ignored it and keep using `print` in later codebase :-D

Reply:
I am trying to use this `logger` but the message no longer show up in the console. I am pretty sure there is a way to do so, but I will just use `print` for now.